src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/DynamicFiles/BackgroundDocumentGenerator.cs

```diff
@@ -55,6 +55,7 @@ protected BackgroundDocumentGenerator(
 
         _disposeTokenSource = new();
         _workQueue = new AsyncBatchingWorkQueue<(IProjectSnapshot, IDocumentSnapshot)>(delay, ProcessBatchAsync, _disposeTokenSource.Token);
+        _workQueue.Idle += (_, _) => RazorEventSource.Instance.BackgroundDocumentGeneratorIdle();
```

Comment:
The event hookup to the lambda here really makes me wonder why an event and `EventHandler` was chosen, since there's no code to unhook the event and the arguments aren't event used.